src/components/mapPageComponents/mapView/index.tsx

```diff
@@ -109,6 +131,49 @@ const MapView: React.FC<MapViewProps> = ({
           },
         });
 
+        class Popup extends google.maps.OverlayView {
```

Comment:
❓ I'm not sure about putting a class definition in the body of a function. This really deserves its own file I think.

Reply:
Yeah, I'll look more into this. The problem with pulling the class out (and most of the code in this hook) is that `google` isn't defined until the DOM loads.